hypothesis-python/src/hypothesis/extra/cli.py

```diff
@@ -82,6 +82,14 @@ def main():
     def obj_name(s: str) -> object:
         """This "type" imports whatever object is named by a dotted string."""
         s = s.strip()
+        if "/" in s:
+            raise click.UsageError(
+                "Remember that the ghostwriter should be passed the name of a module, not a path."
+            )
+        if s.endswith(".py"):
+            raise click.UsageError(
+                "Remember that the ghostwriter should be passed the name of a module, not a file."
+            )
```

Comment:
Let's move this to the last `except ImportError` clause (currently line 109), so that if you *do* happen to have e.g. `def py(x):` in your script that will still work.  Oh, and `raise ... from None` then so that we don't get too long a traceback.

Reply:
This is a great edge case I hadn't thought of. In implementing, I realized there are 2 different places we should be ready to raise an exception for file names ending in ".py":
1. As you suggested, within the last `except ImportError` block, in case the script simply doesn't exist. (Worth noting, calling `obj_name` on a nonexistent filename with only 1 "." [E.g, `myscript.py`] will raise a `ValueError` due to the 2 calls to `rsplit`.)
2. In the event the script _can_ be imported, but doesn't contain a function named "py". Giving an updated exception for this case requires raising within the lower `except AttributeError`. This was the only place I chose to provide a suggestion for the correct invocation, since the other location, the script isn't imported because it likely doesn't exist.